pkg/executables/kubectl.go

```diff
@@ -318,6 +318,24 @@ func (k *Kubectl) ValidateNodes(ctx context.Context, kubeconfig string) error {
 	return nil
 }
 
+func (k *Kubectl) DeleteOldWorkerNodeGroup(ctx context.Context, md *clusterv1.MachineDeployment, kubeconfig, providerName string) error {
+	kubeadmConfigTemplateName := md.Spec.Template.Spec.Bootstrap.ConfigRef.Name
+	providerMachineTemplateName := md.Spec.Template.Spec.InfrastructureRef.Name
+	params := []string{"delete", fmt.Sprintf("machinedeployments.%s", clusterv1.GroupVersion.Group), md.Name, "--kubeconfig", kubeconfig, "--namespace", constants.EksaSystemNamespace}
+	if _, err := k.Execute(ctx, params...); err != nil {
+		return err
+	}
+	params = []string{"delete", fmt.Sprintf("kubeadmconfigtemplates.bootstrap.%s", clusterv1.GroupVersion.Group), kubeadmConfigTemplateName, "--kubeconfig", kubeconfig, "--namespace", constants.EksaSystemNamespace}
+	if _, err := k.Execute(ctx, params...); err != nil {
+		return err
+	}
+	params = []string{"delete", fmt.Sprintf("%smachinetemplates.infrastructure.%s", providerName, clusterv1.GroupVersion.Group), providerMachineTemplateName, "--kubeconfig", kubeconfig, "--namespace", constants.EksaSystemNamespace}
```

Comment:
Do we need the provider name? The machine deployment should have the type it references right? So we maybe extract it from there to pass it to kubectl instead of passing the provider name.